Quadriga/src/main/java/edu/asu/spring/quadriga/service/workbench/StatisticsSettingsService.java

```diff
@@ -0,0 +1,103 @@
+package edu.asu.spring.quadriga.service.workbench;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.asu.spring.quadriga.dao.workbench.settings.IStatisticSettingsDAO;
+import edu.asu.spring.quadriga.domain.IStatisticsSettings;
+import edu.asu.spring.quadriga.dto.StatisticsSettingsDTO;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.mapper.StatisticsSettingsDTOMapper;
+
+/**
+ * this class holds the list of statistics settings for a project and used in
+ * the controller as a modelattribute
+ * 
+ * @author Ajay Modi
+ *
+ */
+@Service
+@PropertySource(value = "classpath:/settings.properties")
+public class StatisticsSettingsService {
+
+    @Autowired
+    private IStatisticSettingsDAO statisticSettingsDAO;
+
+    @Autowired
+    private StatisticsSettingsDTOMapper statisticSettingsDTOMapper;
+
+    @Autowired
+    private Environment env;
+
+    @Transactional
+    public void addOrUpdateStatisticsSettings(String projectId, String[] names)
+            throws QuadrigaStorageException {
+        List<String> defaultValues = getDefaultStatisticsSettingsTypes();
+        List<String> namesList = Arrays.asList(names);
+        for (String name : defaultValues) {
+            StatisticsSettingsDTO statisticsSettingsDTO = statisticSettingsDAO
+                    .getStatisticsSettings(projectId, name);
+            if (namesList.contains(name)) {
+                if (statisticsSettingsDTO == null) {
+                    statisticsSettingsDTO = new StatisticsSettingsDTO(

```

Comment:
you can pull this if statement out into line 47. and then just set set  ischecked.